### PR TITLE
Modernize codebase with Clang-Tidy range based loops, part 3

### PR DESCRIPTION
**Pull request by Lillecarl**
**Changes proposed:**
-  Range based loops everywhere

**Target branch(es):**
- [x] 3.3.5
- [ ] master

**Issues addressed:** 
Keeping TrinityCore up2date with the C++ standards

**Tests performed:**
Builds locally with clang

**Known issues and TODO list:** (add/remove lines as needed)
- [ ] Finish the rest

### CODE REVIEW

File: src/server/game/AuctionHouseBot/AuctionHouseBot.cpp [hunk 1]
```diff
@@ -479,8 +479,8 @@ void AuctionHouseBot::PrepareStatusInfos(AuctionHouseBotStatusInfo& statusInfo)
     {
         statusInfo[i].ItemsCount = 0;
 
-        for (int j = 0; j < MAX_AUCTION_QUALITY; ++j)
-            statusInfo[i].QualityInfo[j] = 0;
+        for (uint32& j : statusInfo[i].QualityInfo)
+            j = 0;
```
Comment (Shauren) [inline src/server/game/AuctionHouseBot/AuctionHouseBot.cpp]: thats a terrible transformed name, something like `countPerQuality` (see next loop for what it does) would have been better

File: src/server/game/Battlefield/Battlefield.cpp [hunk 2]
```diff
@@ -165,14 +165,14 @@ bool Battlefield::Update(uint32 diff)
         if (m_uiKickDontAcceptTimer <= diff)
         {
             time_t now = GameTime::GetGameTime();
-            for (uint8 team = 0; team < PVP_TEAMS_COUNT; ++team)
-                for (PlayerTimerMap::iterator itr = m_InvitedPlayers[team].begin(); itr != m_InvitedPlayers[team].end(); ++itr)
+            for (PlayerTimerMap& m_InvitedPlayer : m_InvitedPlayers)
```
Comment (Shauren) [inline src/server/game/Battlefield/Battlefield.cpp]: another bad transform (m_)

File: src/server/game/Battlefield/Battlefield.cpp [hunk 3]
```diff
@@ -202,8 +202,8 @@ bool Battlefield::Update(uint32 diff)
 
 void Battlefield::InvitePlayersInZoneToQueue()
 {
-    for (uint8 team = 0; team < PVP_TEAMS_COUNT; ++team)
-        for (auto itr = m_players[team].begin(); itr != m_players[team].end(); ++itr)
+    for (GuidUnorderedSet& player : m_players)
```
Comment (Shauren) [inline src/server/game/Battlefield/Battlefield.cpp]: bad auto name - `player` is a container, should be `teamPlayers` or something

File: src/server/game/Battlefield/Battlefield.cpp [hunk 4]
```diff
@@ -961,9 +961,9 @@ bool BfCapturePoint::Update(uint32 diff)
     {
         float radius = capturePoint->GetGOInfo()->capturePoint.radius;
 
-        for (uint8 team = 0; team < PVP_TEAMS_COUNT; ++team)
+        for (GuidSet& m_activePlayer : m_activePlayers)
```
Comment (Shauren) [inline src/server/game/Battlefield/Battlefield.cpp]: bad name, m_

File: src/server/game/Battlefield/Battlefield.cpp [hunk 5]
```diff
@@ -1076,9 +1076,9 @@ bool BfCapturePoint::Update(uint32 diff)
 
 void BfCapturePoint::SendUpdateWorldState(uint32 field, uint32 value)
 {
-    for (uint8 team = 0; team < PVP_TEAMS_COUNT; ++team)
-        for (GuidSet::iterator itr = m_activePlayers[team].begin(); itr != m_activePlayers[team].end(); ++itr)  // send to all players present in the area
-            if (Player* player = ObjectAccessor::FindPlayer(*itr))
+    for (GuidSet& activePlayers : m_activePlayers)
```
Comment (Shauren) [inline src/server/game/Battlefield/Battlefield.cpp]: meh, looking back at all transformations like this one I'm not sure that this brings any improvement... you now have to give `m_activePlayers[team]` a name (TYPE activePlayers : m_activePlayers tells nothing to the reader)

**Comment by Shauren** [comment]
and i didnt even check how many of these can be made const

File: src/server/game/AuctionHouseBot/AuctionHouseBot.cpp (hunk 1 above)
Comment (Lillecarl) [reply src/server/game/AuctionHouseBot/AuctionHouseBot.cpp]: true.

**Comment by illfated** [comment]
Looks like AppVeyor complains about your removal of line 42 in `AppenderConsole.h` (https://github.com/TrinityCore/TrinityCore/pull/24228/files#diff-a1e7b12cc418f35998e2cdf42c5334d5L42)

C:\projects\trinitycore\src\common\Logging\AppenderConsole.cpp(69): error C2065:
` 'MaxColors': undeclared identifier` [C:\projects\trinitycore\build\src\common\common.vcxproj]

C:\projects\trinitycore\src\common\Logging\AppenderConsole.cpp(72): error C2078:
` too many initializers` [C:\projects\trinitycore\build\src\common\common.vcxproj]

**Comment by jackpoz** [comment]
any update on this PR ?

**Comment by jackpoz** [comment]
Closing this PR as it seems abandoned. Feel free to ask here to reopen it if anyone would like to work on it.

**Pull request closed without merging**